src/sentry/utils/registry.py

```diff
@@ -15,9 +15,10 @@ class NoRegistrationExistsError(ValueError):
 
 
 class Registry(Generic[T]):
-    def __init__(self):
+    def __init__(self, enable_reverse_lookup=True):
```

Comment:
Might be worth a docstring explaining that it's useful to disable reverse lookup when you have duplicate values